packages/framework-provider-local/src/library/web-socket-server-adapter.ts

```
@@ -0,0 +1,95 @@
+  private readonly httpServer: HttpServer
+  private readonly _webSocketServer: Server
+  private readonly clients: Record<string, WebSocket> = {}
+  private readonly WS_PORT = '65529'
+
+  constructor(readonly graphQLService: GraphQLService, config: BoosterConfig) {
+    const expressServer = express()
+    this._webSocketServer = new WebSocket.Server({
+      noServer: true,
+      path: '/websockets',
+    })
+    this.httpServer = expressServer.listen(this.WS_PORT, () => {})
+    const logger = getLogger(config, 'WebSocketServerAdapter#constructor#handleUpgrade')
+    logger.info(`Starting websocket server on ws://localhost:${this.WS_PORT}/websockets`)
+    this.attach()
+    this.onConnection()
+  }
+
+  public attach(): void {
+    this.httpServer.on('upgrade', (request: IncomingMessage, socket: Duplex, head: Buffer) => {
+      this._webSocketServer.handleUpgrade(request, socket, head, (webSocket: WebSocket) => {
+        this._webSocketServer.emit('connection', webSocket, request)
+      })
+    })
+  }
+
+  private onConnection(): void {
+    this._webSocketServer.on('connection', async (webSocket: WebSocket, connectionRequest: IncomingMessage) => {
+      const connectionId = UUID.generate()
+      this.clients[connectionId.toString()] = webSocket
+      const request: ExpressWebSocketMessage = {
+        incomingMessage: connectionRequest,
+        connectionContext: {
+          connectionId: connectionId,
+          eventType: 'CONNECT',
+        },
+      }
+      await this.graphQLService.handleGraphQLRequest(request)
+      this.onMessage(webSocket, connectionRequest, request.connectionContext.connectionId)
+      this.onDisconnect(webSocket, request.connectionContext.connectionId)
+    })
+  }
+
+  private onMessage(webSocket: WebSocket, connectionRequest: IncomingMessage, connectionId: UUID) {
+    webSocket.on('message', async (message) => {
+      const request: ExpressWebSocketMessage = {
+        data: JSON.parse(message.toString()),
```

This JSON.parse can actually fail in the case the string is not a proper JSON, let's better have a try/catch here

---

I update the code to ignore the error, but let me know what you think about it